src/codegen/unwinding.cpp

```diff
@@ -502,6 +519,36 @@ static const LineInfo* lineInfoForFrame(PythonFrameIteratorImpl& frame_it) {
     return new LineInfo(current_stmt->lineno, current_stmt->col_offset, source->fn, source->getName());
 }
 
+// To produce a traceback, we:
+//
+// 1. Use libunwind to produce a cursor into our stack.
+//
+// 2. Grab the next frame in the stack and check what function it is from. There are four options:
+//
+//    (a) A JIT-compiled Python function.
+//    (b) ASTInterpreter::execute() in codegen/ast_interpreter.cpp.
+//    (c) generatorEntry() in runtime/generator.cpp.
+//    (d) Something else.
+//
+//    By cases:
+//
+//    (2a, 2b) If the previous frame we visited was an OSR frame (which we know from its CompiledFunction*), then we
+//    skip this frame (it's the frame we replaced on-stack) and keep unwinding. (FIXME: Why are we guaranteed that we
+//    on-stack-replaced at most one frame?) Otherwise, we found a frame for our traceback! Proceed to step 3.
```

Comment:
well, at least in our current implementation, OSRs only replace the top-most frame.  we can have multiple OSR'd frames in a row, such as 
1: interpreter frame
2: interpreter->lowest level jit OSR
3: jit -> optimized jit OSR

frames 2 and 3 will have the "is osr" bit set on them, which should mean that frames 1 and 2 get excluded from the traceback.  so in theory we should get one traceback entry per non-osr frame.